### ISSUE TITLE
more detail on the vm checks - not every stopped machine is a warning

### ISSUE THREAD

**Issue by johnnychicago** [nbuchwitz/check_pve#6]
Hello --

I'd like to be able to configure which vm's should give back warning or critical statuses if not running - some vm's may be ran only sporadically for a reason. I'd like to track them using icinga and grafana, but I am not bothered if they're off. Any way to have a (per vm) variable to describe this?

Thanks

**Comment by nbuchwitz** [comment]
Hi,

I added an flag named ``--ignore-vm-status`` which is used in the check result calculation:

```
./check_pve.py -e pve -u monitor@pve -p secret --node pve --mode vm --name fedora

CRITICAL - VM 'fedora' not running
```

```
./check_pve.py -e pve -u monitor@pve -p secret --node pve --mode vm --name fedora --ignore-vm-status

OK - VM 'fedora' not running
```

Happy to hear if it's working for you!

**Comment by johnnychicago** [comment]
Hey - thanks a lot for looking at this. Not close to a cli right now to try this, will feed back later.

Conceptually, might be nice to give it some more​ flexibility by specifying​ what status a not running vm triggers. Some would be warnings for me, some would be  irrelevant, other users may consider some critical. But then, maybe I'm overcomplicating :-)

Cheers

On 14 July 2018 12:05:27 CEST, Nicolai <notifications@github.com> wrote:
>Hi,
>
>I added an flag named ``--ignore-vm-status`` which is used in the check
>result calculation:
>
>```
>./check_pve.py -e pve -u monitor@pve -p secret --node pve --mode vm
>--name fedora
>
>CRITICAL - VM 'fedora' not running
>```
>
>```
>./check_pve.py -e pve -u monitor@pve -p secret --node pve --mode vm
>--name fedora --ignore-vm-status
>
>OK - VM 'fedora' not running
>```
>
>Happy to hear if it's working for you!
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/nbuchwitz/check_pve/issues/6#issuecomment-405013119

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by nbuchwitz** [comment]
No problem, I think an expected status would make sense in some scenarios. I added another parameter ``--expected-vm-status`` which can be either ``running`` or ``stopped``.

For complexity reasons I wouldn't like to make the return status configurable between ``WARNING`` and ``CRITICAL``. So a VM in a non expected state (which is per default ``running``) will produce a ``CRITICAL`` state. The ``WARNING`` state will be returned if the VM couldn't be found on the cluster.

**Comment by johnnychicago** [comment]
OK, that took a bit to actually click for me - I need to fiddle with a combination of both flags to get where I want to go. I can work with that, although it's not as intuitive to me.

Thanks a lot!

BTW - VM's can be in a paused/suspended state as well, not just running or stopped. Just sayin' ;)

**Comment by johnnychicago** [comment]
minor thing - in icinga2/command.conf, line 50, I replaced value with set_if to make the ignore-vm-status flag work.
